src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -326,13 +327,21 @@ def window(
             end: end time of the window.
             force_copy: optional, default True. If True, will always create a
                 copy of the data.
+            fill_value: optional. If not None, will use this value to fill
+                missing values. If missing values should be filled, force_copy
+                must be True. Defaults to NaN to avoid returning outdated data
+                unexpectedly.
 
         Raises:
             IndexError: When start and end are not both datetime or index.
+            ValueError: When fill_value is not None and force_copy is False.
 
         Returns:
             The requested window
         """
+        if fill_value is not None and not force_copy:
+            raise ValueError("Fill NA only supported for force_copy=True")
```

Comment:
out of curiosity, what is it `NA`?

Reply:
I had the option named `fill_na` before, adapted from pandas. NA indicates missing values `None` and `NaN`. Renamed it to make it more descriptive. Fixed this in the update.